### ISSUE TITLE
Plug-Ins from various developers stop working and crashing in FCP

### ISSUE THREAD

**Issue by stevef243** [CommandPost/FCPCafe#296]
Apple Feedback Assistant ID: FB13453542
---

**DESCRIBE THE BUG:**
Plug-Ins from various developers stop working and crashing in FCP often this is marked with a Pop up window that thge PlugIn stopped responding but there are also quite a few occasions where the pop-up does not appear, especially during export, and then you have the read missing screen in the export, 

I have noticed this is quite frequent see the same issue on other content creators' videos as there is no FCP warning that something is missing. It seems like the longer FCP is open, the more likely the PluG Ins fail to respond. A restart for FCP fixes for a little while.

This started after the move to Apple Silicon

---

**TO REPRODUCE:**
use PlugIns and leave FCP open for a while

---

**EXPECTED BEHAVIOUR:**
PlugIns continue to work, warning message if exports have issues

---

**SCREENSHOTS:**

![CleanShot 2023-06-18 at 15 34 45@2x](https://github.com/CommandPost/FCPCafe/assets/152864795/c11479ac-379e-45f4-a9fe-ba21f4e0c994)
![CleanShot 2023-10-25 at 12 55 00@2x](https://github.com/CommandPost/FCPCafe/assets/152864795/cc4d9609-1804-456e-b1bf-899f328ef413)
![CleanShot 2023-11-01 at 16 11 41@2x](https://github.com/CommandPost/FCPCafe/assets/152864795/326cc337-dd61-4426-80a0-4a3c17c77115)

---

**SPECS:**
- Mac Studio
- macOS Sonoma
- Final Cut Pro 10.6, 7.x

**Comment by stevef243** [comment]
As I mentioned not many in there and non which seem to be related to the crashes

I will have a look at the activity monitor next time it happens currently not working on a larger project with plug ins

**Comment by latenitefilms** [comment]
For the avoidance of doubt, and for anyone who randomly comes across this GitHub issue, whenever you get a **A plug-in is not responding** alert - this means that a plugin has crashed - either due to a bug in the FxPlug4 code, or because FCPX has correctly or incorrectly "force quit" it.

In both cases however, there will always be a crash log for these crashes.

Where to find crash logs is explained here: https://fcp.cafe/bugtracker/#finding-crash-logs

For example:

<img width="1097" alt="image" src="https://github.com/CommandPost/FCPCafe/assets/22286696/c0c3b621-13f1-449f-95dd-b19cd0b9d3c0">

FWIW - in the crash logs above, please ignore the sheer quantity of Gyroflow, and Metaburner Toolbox crashes, as this is my development machine, so I'm constantly crashing things whilst coding.

**Comment by stevef243** [comment]
Put it like this: I never cleared the crash log, and I can tell you there should be dozens of crashes in there, as I often did screenshots.
Of the crashed module in FCP. So unless Apple removes them, they are not there.
Bus, as described, while FCP is also crashing occasionally, the main issue is all sorts of Plug-ins crashing and Apple not being able to restart them.
Often that gives the mentioned error message but often also not especially during export

**Comment by latenitefilms** [comment]
The crash logs do get uploaded frequently. They move into the Retired folder, then get deleted. I'm not sure how often it gets uploaded and trashed - maybe each time you logout of your Mac?

macOS should 100% definitely generate a crash log every time you see **A plug-in is not responding** for an FxPlug4-based effect. 

If it's a straight Motion Template (i.e something made entirely in Apple Motion), or an Audio Unit (i.e. an Audio Effect) then you won't see a crash - but for anything that uses FxPlug4, you definitely will.

It's also entirely possible, if you turned off **Share Mac Analytics** or **Share with app developers** in System Settings, it MIGHT have an impact on the log files actually being written to disk - but I'm not 100% sure of that:

<img width="715" alt="image" src="https://github.com/CommandPost/FCPCafe/assets/22286696/adbc7e6c-5da3-4cb6-b402-4c173d39e58b">

Another thing you can try is holding down **OPTION** when you click the the **Help** menubar item in Final Cut Pro, and you'll see a **Gather App Diagnostics** option appear. When you click this, wait 5 minutes, then you'll see a file with a name like **VideoAppDiagnostics-FinalCut-2023-12-08-083149.tar** appear on your Desktop. This should also contain any crash logs.

**Comment by latenitefilms** [comment]
Ok, sorry, I mean, you shouldn't disable these items.

**Comment by stevef243** [comment]
ok re-enabled this seems to show a few crashes option/help
[system_log-2023-12-08-083901.log.zip](https://github.com/CommandPost/FCPCafe/files/13605740/system_log-2023-12-08-083901.log.zip)

**Comment by stevef243** [comment]
I'm no developer, but tons of Apple Runtime issue.

2023-12-05 09:30:46.466491+1100 0x14fdc    Fault       0xbcc01              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:30:46.506327+1100 0x14fdc    Fault       0xbcc02              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:31:01.102058+1100 0x14fdc    Fault       0xbcc03              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:31:05.147311+1100 0x14fdc    Fault       0xbcc04              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:31:06.975201+1100 0x14fdc    Fault       0xbcc05              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:31:07.024967+1100 0x14fdc    Fault       0xbcc06              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:31:15.456382+1100 0x14fdc    Fault       0xbcc09              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:31:15.726184+1100 0x14fdc    Fault       0xbcc0a              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:31:18.256338+1100 0x14fdc    Fault       0xbcc0b              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:33:39.068152+1100 0x14fdc    Fault       0xc65c0              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:33:39.085637+1100 0x14fdc    Fault       0xc65c1              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:33:44.132290+1100 0x14fdc    Fault       0xc65c2              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:33:44.381038+1100 0x14fdc    Fault       0xc65c3              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>
2023-12-05 09:33:46.375096+1100 0x14fdc    Fault       0xc65c4              5945   14   Final Cut Pro: (AppKit) [com.apple.runtime-issues:Nib Loading] Could not connect action, target class FFTransform2DOSC does not respond to -<private>

**Comment by latenitefilms** [comment]
You can ignore all those messages - they're just developer noise.

**Comment by stevef243** [comment]
There are definitely no diagnostic reports for me, and this is becoming ridiculous. 

https://drive.google.com/file/d/1tc-Y38tQldf403WG4OkBTkmiPrjdp4li/view?usp=sharing
![CleanShot 2023-12-11 at 15 10 34@2x](https://github.com/CommandPost/FCPCafe/assets/152864795/a6ddaf2c-196e-45c0-b525-ef3e4610b824)
![CleanShot 2023-12-11 at 15 12 25@2x](https://github.com/CommandPost/FCPCafe/assets/152864795/1db5821c-6e46-454c-917d-f292913afd1d)

**Comment by latenitefilms** [comment]
I'm very sorry to hear you're still having issues! That really does suck.

What Motion Template is failing? Is it always the same Motion Template or does it happen to multiple ones?

My only GUESS as to the reason you're not seeing crash logs is because it's happening during the export, not in the timeline during normal editing. I'm not really sure how exporting works, but it's entirely possible that when a FxPlug crashes in Final Cut Pro during normal editing, you get a crash log - but in this case it seems to be silently crashing (or at least failing) during the export process, which might be triggered by a background process.

Do you have background render enabled? It's hard to tell from the video - is your timeline all rendered?

If so, what happens if you disable background render, and delete all the render files. Does it export ok now?

Alternatively, if you have background render disabled, and nothing rendered, what happens if you render and export?

What happens if you Send to Compressor and export there?

Are you exporting a H.264/HEVC? If so, what happens if you export a ProRes file instead?

**Comment by stevef243** [comment]
This time it's Stupid Raisins Side Pop from FX Factory I find developer does not matter.

No disabled background render a while ago on advice. I re-started Final Cut Pro and re-rendered, and it now exported fine, but there are quite a few extra steps, which really should not be required.

![CleanShot 2023-12-11 at 15 54 40@2x](https://github.com/CommandPost/FCPCafe/assets/152864795/7b89d21f-b005-4dfb-a892-39a79bbca74e)

**Comment by latenitefilms** [comment]
Pinging @sharedscene - are any other FxFactory users seeing similar things? Could it be a matter of the Motion Templates needing to be updated to the latest Motion version? Based on the copyright, it looks like this plugin hasn't been updated in a loooooooong time?

**Comment by stevef243** [comment]
as mentioned also happens with MotionVFX plug ins, usually if you have FCP open for a while even in the background

**Comment by latenitefilms** [comment]
Do you get the "plugin not responding" message during the export, or does it just play fine on the timeline, and export without any issues, but then when you review, you noticed that plugins are broken?

Can you share a Diagnostics Bundle? This is explained on the [Bug Tracker page](https://fcp.cafe/bugtracker/#final-cut-pro-diagnostics-bundle).

**Comment by stevef243** [comment]
yes that's what happened with this one no warning whatsoever

**Comment by stevef243** [comment]
https://drive.google.com/file/d/1wryC03h3MeQRE_XOIVeyPvjG0JqwO-jG/view?usp=sharing

**Comment by latenitefilms** [comment]
Ok, so this is the hardware specs for easy future reference:

**Model Name:** Mac Studio
**Model Identifier:** Mac13,1
**Model Number:** Z14J0006UX/A
**Chip:** Apple M1 Max
**Total Number of Cores:** 10 (8 performance and 2 efficiency)
**Memory:**	32 GB

---

Some interesting observations...

This error in `SystemLog` might be releated to your plugins disappearing (`Ozone` is the Apple Motion Engine):

```
Final Cut Pro: (Ozone) Unable to setup plugin connection in createPluginWithClass_block_invoke due to error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service created from an endpoint was invalidated: failed to check-in, peer may have been unloaded: mach_error=10000003." UserInfo={NSDebugDescription=The connection to service created from an endpoint was invalidated: failed to check-in, peer may have been unloaded: mach_error=10000003.}
Final Cut Pro: (Ozone) Error creating plugin with class name "FxFactoryFxFilter_FC1B95ECC419452783D364599FD9EDC2" : (null)
Final Cut Pro: (Ozone) Unable to setup plugin connection in createPluginWithClass_block_invoke due to error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service created from an endpoint was invalidated from this process." UserInfo={NSDebugDescription=The connection to service created from an endpoint was invalidated from this process.}
 Final Cut Pro: (Ozone) Error creating plugin with class name "FxFactoryFxGenerator_B886DB4FBD7442FA95200A41F8993764" : (null)
```

Googling around, this comment is really interesting:

> In my case I solved the issue. The issue was related to a bundle Id of an app I have in the AppStore. Using that specific bundle Id in any app with NSOpenPanel (or NSSavePanel) resulted inthe error. Replace the bundle Id in the same project with another one and the problem goes away. Why? Some caching issue of XCode. Cleaning the build folder did not help. So I did a full uninstall and removal of all directories related to XCode. Then reinstalled XCode. No more problem. I'm thinking perhaps an outdated entry TCC.db?

Could you try resetting the TCC database and see if that has any impact?

You can reset it using this Terminal command: `tccutil reset All`

This will reset all your **Privacy & Security** Settings, so you'll be prompted for permissions again by all your apps, which is annoying - but it might fix your issue.

Curious... did you upgrade from a previous macOS release to Sonoma?

**Related:** 
- [A deep dive into macOS TCC.db](https://www.rainforestqa.com/blog/macos-tcc-db-deep-dive)
- [macOS TCC](https://book.hacktricks.xyz/macos-hardening/macos-security-and-privilege-escalation/macos-security-protections/macos-tcc)

---

I also spotted this:

```
Final Cut Pro: (Ozone) Error creating plugin with class name "mUtility" : (null)
```

I also saw this:

```
2023-12-11 09:30:56.856528+1100 0x1843d    Error       0x0                  3135   0    Final Cut Pro: (CoreFoundation) [com.apple.CFBundle:loading] Attempt to load executable of a type that cannot be dynamically loaded for CFBundle 0x600003cc33a0 </Applications/motionVFX/Plugins/mUtility.app/Contents/PlugIns/mUtility XPC Service.pluginkit> (executable, not loaded)
```

I assume you've already tried making sure all your MotionVFX plugins are up-to-date and that you've done all MotionVFX's various checks in mInstaller?

<img width="1191" alt="image" src="https://github.com/CommandPost/FCPCafe/assets/22286696/a1d39119-ce6b-41ee-9903-1e4c4c9958e0">

Looking for more mUtility lines:

```
2023-12-11 08:49:38.455440+1100 0xf13c     Default     0x3bd79              3135   0    Final Cut Pro: (RunningBoardServices) [com.apple.runningboard:process] Hit the server for a process handle 529da1a00000c4a that resolved to: [xpcservice<mUtility-XPC-Service([app<application.com.apple.FinalCut.50092462.50095991(501)>:3135])(501)>:3146]
2023-12-11 08:49:38.455467+1100 0xf13c     Default     0x3bd79              3135   0    Final Cut Pro: (libxpc.dylib) [com.apple.xpc:connection] [0x12e68d4c0] activating connection: mach=false listener=false peer=false name=mUtility-XPC-Service
2023-12-11 09:30:56.848009+1100 0x1843d    Error       0x0                  3135   0    Final Cut Pro: (CoreFoundation) [com.apple.CFBundle:loading] Attempt to load executable of a type that cannot be dynamically loaded for CFBundle 0x600003cc33a0 </Applications/motionVFX/Plugins/mUtility.app/Contents/PlugIns/mUtility XPC Service.pluginkit> (executable, not loaded)
2023-12-11 09:30:56.852222+1100 0x1843d    Error       0x0                  3135   0    Final Cut Pro: (CoreFoundation) [com.apple.CFBundle:loading] Attempt to load executable of a type that cannot be dynamically loaded for CFBundle 0x600003cc33a0 </Applications/motionVFX/Plugins/mUtility.app/Contents/PlugIns/mUtility XPC Service.pluginkit> (executable, not loaded)
2023-12-11 09:30:56.856170+1100 0x1843d    Error       0x0                  3135   0    Final Cut Pro: (CoreFoundation) [com.apple.CFBundle:loading] Attempt to load executable of a type that cannot be dynamically loaded for CFBundle 0x600003cc33a0 </Applications/motionVFX/Plugins/mUtility.app/Contents/PlugIns/mUtility XPC Service.pluginkit> (executable, not loaded)
```

---

Digging deeper...

Lots of logs lines with `AUHostingServiceClient` crashing:

```
2023-12-10 21:13:21.722708+1100 0xc6771    Error       0x0                  9396   0    Final Cut Pro: (AudioToolboxCore) [com.apple.coreaudio:AUHostingService] AUHostingServiceClient.mm:231   [AUHostingService Client] connection invalidated.
```

LOTS of this:

```
Final Cut Pro: (libxpc.dylib) [com.apple.xpc:connection] [0x46dce7310] invalidated after a failed send of the check-in message: mach_error=[0x10000003: (ipc/send) invalid destination port]
```

Some other interesting lines in the `SystemLog`:

```
Final Cut Pro: (libAudioStatistics.dylib) [com.apple.coreaudio:carc]      CAReportingClient.mm:532   Attempted to remove a reporter not created by this client { careporter_id=40355512713275 }
Final Cut Pro: (CoreMediaIO) [com.apple.cmio:] CMIOHardware.cpp:295:CMIOObjectGetPropertyData the System is exiting
Final Cut Pro: (CoreMediaIO) [com.apple.cmio:] CMIOHardware.cpp:331:CMIOObjectGetPropertyData Error: 1970171760, failed
Final Cut Pro: (CoreMediaIO) [com.apple.cmio:] CMIO_DAL_PlugInManagement.cpp:917:CreatePlugIn Could not find plugin with kCMIOHardwarePlugInTypeID
 Final Cut Pro: (CoreAudio) AudioHardware-mac-imp.cpp:1224   AudioObjectRemovePropertyListener: no object with given ID 109
 Final Cut Pro: (PlugInKit) [com.apple.PlugInKit:xpc] <private>: XPC error talking to pkd: Connection interrupted
 Final Cut Pro: (CoreAudio)      HALC_ShellPlugIn.cpp:860    HAL_HardwarePlugIn_ObjectHasProperty: no object
 Final Cut Pro: (CoreAudio) AudioHardware-mac-imp.cpp:1224   AudioObjectRemovePropertyListener: no object with given ID 247
 Final Cut Pro: (libAudioStatistics.dylib) [com.apple.coreaudio:carc]      CAReportingClient.mm:532   Attempted to remove a reporter not created by this client { careporter_id=13464722473012 }
 Final Cut Pro: (CoreMedia) [com.apple.coremedia:] <<<< FigCustomURLHandling >>>> curlh_finishLoadingOnQueue: com.apple.avfoundation.customurl.cfurlconnection.0x60001bf44300: requestID: 2, error: Error Domain=kCFErrorDomainCFNetwork Code=-1002 "unsupported URL" UserInfo={NSErrorFailingURLStringKey=mediaserver:///var/folders/mh/y95kb8_56x5bw1pz0ls0sxnw0000gn/T/1C0E34F4-B38B-4987-9FEC-1B1C79A60E60.export-seq-id?itemname=3978CE10-6A0D-42AA-851E-445C5967F632@23F62815-8D2B-4454-82A4-487BC4FB0D75%26framerate=29.97003%26in=0%26out=35107%26nativeCodec=flexo%26nativeCodecMfr=apcn%26displayName=52656461726320416C70686120313530%26width=3840%26height=2160%26startTC=R30000/1001%200%20DF%26fieldDom=1%26channelLayout=6619138%26sampleRate=48000%26providerclass=ProMSMediaServer%26paspV=1%26paspH=1%26raw-color-primaries=1%26raw-color-transfer=1%26raw-color-matrix=1, NSLocalizedDescription=unsupported URL, NSErrorFailingURLKey=mediaserver:///var/folders/mh/y95kb8_56x5bw1pz0ls0sxnw0000gn/T/1C0E34F4-B38B-4987-9FEC-1B1C79A60E60.export-seq-id?itemname=3978CE10-6A0D-42AA-851E-445C5967F632@23F62815-8D2B-4454-82
 Final Cut Pro: (CoreMedia) [com.apple.coremedia:] <<<< FigCustomURLHandling >>>> curlh_finishLoadingOnQueue: com.apple.avfoundation.customurl.cfurlconnection.0x60001304d5c0: requestID: 2, error: Error Domain=kCFErrorDomainCFNetwork Code=-1002 "unsupported URL" UserInfo={NSErrorFailingURLStringKey=mediaserver:///var/folders/mh/y95kb8_56x5bw1pz0ls0sxnw0000gn/T/1C0E34F4-B38B-4987-9FEC-1B1C79A60E60.export-seq-id?itemname=3978CE10-6A0D-42AA-851E-445C5967F632@23F62815-8D2B-4454-82A4-487BC4FB0D75%26framerate=29.97003%26in=0%26out=35107%26nativeCodec=flexo%26nativeCodecMfr=apcn%26displayName=52656461726320416C70686120313530%26width=3840%26height=2160%26startTC=R30000/1001%200%20DF%26fieldDom=1%26channelLayout=6619138%26sampleRate=48000%26providerclass=ProMSMediaServer%26paspV=1%26paspH=1%26raw-color-primaries=1%26raw-color-transfer=1%26raw-color-matrix=1, NSLocalizedDescription=unsupported URL, NSErrorFailingURLKey=mediaserver:///var/folders/mh/y95kb8_56x5bw1pz0ls0sxnw0000gn/T/1C0E34F4-B38B-4987-9FEC-1B1C79A60E60.export-seq-id?itemname=3978CE10-6A0D-42AA-851E-445C5967F632@23F62815-8D2B-4454-82
```

Hopefully someone from Apple is listening and can provide some better insight...

**Comment by stevef243** [comment]
did again the the mInstaller checks and repairs, yes everything is up to date.
I just reset TCC lets see if that helps hop all apps ask for permissions again?

**Comment by latenitefilms** [comment]
Yes, after running `tccutil reset All` it will reset all your **Privacy & Security** Settings, so you'll be prompted for permissions again by all your apps. Probably worth doing a shut down and restart after running `tccutil reset All` just for good measure.

Fingers and toes crossed that helps!

**Comment by latenitefilms** [comment]
@stevef243 - FYI - Apple has reached out to me about this, and are actively investigating.

**Comment by stevef243** [comment]
wow not sure what connections you have but that’s unheard of, just posted a new one going on since 10.7 where FCp hangs when updating libraries. Have easy +50 Libraries and this update business for every update is painful especially when FCP hangs. Had this happen in one out of six libraries. After a force quite FCP restart the updated library looks ok

**Comment by latenitefilms** [comment]
I'm kinda hoping someone from Apple might actually respond here, but it seems unlikely - Apple staff generally aren't allowed to reply on public forums sadly, for whatever reason. 

I also don't think I should share their exact response publicly, but just reading between the lines, here's my PERSONAL opinion...

Because you're (so far), the only person that's really reported this, there's not much for them to go on. If more people reported the same issue it would be far easier for them to fault-find.

Based on all of the above, my GUESS is that it's actually macOS that's "killing" the plugins due to memory getting low - which is why you're not seeing crash logs. If macOS kills it - it's not technically a crash. Because FxPlug4 plugins now run "out of process" each plugin is essentially their own application - and they use the macOS Extensions system to work - for example, you can see all your FxPlug Extensions in **Privacy & Security > Extensions > Added extensions** (on Sonoma at least):

<img width="715" alt="image" src="https://github.com/CommandPost/FCPCafe/assets/22286696/efc47536-04c5-4282-b033-5b9020bc85d8">

Basically, the whole reason I created this Bug Tracker is for situations like this... Apple needs more information to fault-find the problem, as it clearly doesn't happen to everyone. If we can get the whole FCPX community on-board, and if EVERYONE who had the same issue shared their log files and diagnostics here, then we could probably get a response, and maybe even a fix pretty quickly. But with only one user reporting this, and not really enough information in the log files or diagnostic files to go on, it makes it tricky.

If this happens again, can you also take a screenshot of the Memory screen in Activity Monitor?

**Comment by latenitefilms** [comment]
@stevef243 - Someone at Apple has requested... Could you please run **System Diagnostics** via [Activity Monitor](https://support.apple.com/en-au/guide/activity-monitor/actmntr2225/mac) and upload your results?

<img width="311" alt="image" src="https://github.com/CommandPost/FCPCafe/assets/22286696/dabc2bad-7329-4909-8e08-ef6d67dbd0d8">

Also, just to clarify... my understanding is that in **some cases**, you've been able to edit totally fine in Final Cut Pro - everything is working fine in the timeline, with **no** "this plugin is not responding" alerts. It also exports fine, with **no** "this plugin is not responding" alerts, HOWEVER when you review the export, it has red error frames, signifying that an FxPlug effort failed to render during export. Is that a correct summary?

**Comment by stevef243** [comment]
yes, that is the correct summary.

sys diagnosis link
https://drive.google.com/file/d/18Tw8m-bnovMcGz3p5x22918h0J0lhuSq/view?usp=sharing

**Comment by latenitefilms** [comment]
Legend, thank you! I've passed this on to the ProApps team.

I'll keep you posted if I hear anything back.

I'm assuming `tccutil reset All` didn't fix your problem?

**Comment by sharedscene** [comment]
@stevef243 sorry if you previously mentioned it; what version of FCP & macOS are you on?
how much RAM does your machine have?

we were seeing these issues a lot during the first 2 years of Apple Silicon, often with 8-16GB configurations. Apple had *mostly* fixed it earlier this year with macOS, FCP & FxPlug updates. it was common that crash reports were never generated, because the plugins weren't truly crashing.

**Comment by stevef243** [comment]
MacStudio M1 Max 32GB Ram Sonomoa 14.1.2 FCP 10.7

**Comment by stevef243** [comment]
> Legend, thank you! I've passed this on to the ProApps team.
> 
> I'll keep you posted if I hear anything back.
> 
> I'm assuming `tccutil reset All` didn't fix your problem?

to early to say, sure didn't fix hanging FCP when doing DB updates

**Comment by latenitefilms** [comment]
> to early to say, sure didn't fix hanging FCP when doing DB updates

No worries - next time you do an export, and there's red render error frames, please collect System Diagnostics & Final Cut Pro Diagnostics and post them here again. Hopefully we can work out the cause and Apple can address.

**Comment by stevef243** [comment]
[Archive.zip](https://github.com/CommandPost/FCPCafe/files/13847965/Archive.zip)
now always crashing on export

**Comment by stevef243** [comment]
[Archive.zip](https://github.com/CommandPost/FCPCafe/files/13848079/Archive.zip)

some more crash reports got errer on export stabilisation was not finished had to spend now 30 minutes going trough every clip who was stabilised and re-render now finally can export it's a joke really. Would be very helpfull if FCP could specify what clip exactly has issues in a 30 minute video with hundred of clips doing this manual is a joke

**Comment by stevef243** [comment]
Here is another example of plugins just stopping working with no notification or the like; a FCP restart fixes it. It seems to most likely happen if FCP is open for a while even in the background like it's running out of memory and then stopping addons.

https://drive.google.com/file/d/1z1x3_cghpKJvOuqEO83p2a-qSdsb7hdZ/view?usp=sharing
